src/Functions/array/arrayFirstLastIndex.cpp

```diff
@@ -125,7 +125,44 @@ using FunctionArrayLastIndex = FunctionArrayMapped<ArrayLastIndexImpl, NameArray
 
 REGISTER_FUNCTION(ArrayFirstIndex)
 {
-    factory.registerFunction<FunctionArrayFirstIndex>();
+    FunctionDocumentation::Description description_first = R"(
+Returns the index of the first element in the source array for which `func(x[, y1, y2, ... yN])` returns something other than `0`, otherwise it returns '0'.
+)";
+    FunctionDocumentation::Syntax syntax_first = "arrayFirstIndex(func(x[, y1, ..., yN]), source[, cond1, ... , condN])";
+    FunctionDocumentation::Arguments arguments_first = {
+        {"func(x[, y1, ..., yN])", "A lambda function which operates on elements of the source array (`x`) and condition arrays (`y`). [Lambda function](/sql-reference/functions/overview#arrow-operator-and-lambda)."},
+        {"source", "The source array to process. [`Array(T)`](/sql-reference/data-types/array)."},
+        {"[, cond1, ... , condN]", "Optional. N condition arrays providing additional arguments to the lambda function. [`Array(T)`](/sql-reference/data-types/array)."}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_first = "Returns the index of the first element of the source array for which `λ` is not `0`, otherwise returns `0`. [`UInt32`](/sql-reference/data-types/int-uint).";
+    FunctionDocumentation::Examples examples_first = {
+        {"Usage example", "SELECT arrayFirstIndex(x, y -> x=y, ['a', 'b', 'c'], ['c', 'b', 'a'])", "2"},
+        {"No match", "SELECT arrayFirstIndex(x, y -> x=y, ['a', 'b', 'c'], ['d', 'e', 'f']) ", "0"}
+    };
+    FunctionDocumentation::IntroducedIn introduced_in_first = {1, 1};
```

Comment:
Sure that function was already in v1.1? It sounds like it was introduced later.

Reply:
Checked again and it's v1.1 on the list. Maybe not entirely true but it was added 4 years ago https://github.com/ClickHouse/ClickHouse/pull/33465 and i can't find it in any change logs so let's leave it as 1.1 i think.